07_leukemia_patient/workflow/workflow.sh

```diff
@@ -0,0 +1,232 @@
+#!/bin/bash
+##
+## Giulia Moro
+##
+## Fusion analysis for the leukemia dataset
+##
+
+## this script is to be run after the rock_roi_method workflow: https://github.com/imallona/rock_roi_method/tree/main
+# only on TSO data, since the fusions are expected to be there 
+
+# for STARsolo
+
+COMBINED_FA_GENOME=~/mapping_leukemia_data/genome/combined.fa
+COMBINED_GTF_GENOME=~/mapping_leukemia_data/genome/combined.gtf
+WD=~/test_leukemia_simulated_reads
+COMBINED_INDEXED_GENOME=~/mapping_leukemia/data/index
+STARSOLO_BAM=$WD/starsolo/Aligned.sortedByCoord.out.bam
+r1=~/simulated_leukemia_data/combined_r1.fastq.gz
+r2=~/simulated_leukemia_data/combined_r2.fastq.gz
+r2_no_gz=~/simulated_leukemia_data/combined_r2.fastq
+
+# for bwa mem2
+
+TRANSCRIPTOME=gencode.v38.pc_transcripts.fa
+
+CUSTOM_FA=~/leukemia_bwamem2/genome/BCR_ABL.fa
+CUSTOM_GTF=~/leukemia_bwamem2/genome/BCR_ABL.gtf
+
+# for STAR fusion
+
+STAR_FUSION_GENOME=~/test_starfusion/genome/ctat_genome_lib_build_dir
+PATH_STAR_FUSION=~/STAR-Fusion/ctat-genome-lib-builder
+PATH_SIMG_FILE=~/star-fusion.v1.13.0.simg
+
+# get unmapped reads and reads that mapped to BCR and to ABL and append their name to the readname
+
+cd $WD
+
+# for mapped reads: they will have the gx tag, unmapped won't so will have different column number
+
+# getting reads in BCR and ABL region and adding 1000 bp on each side of the region
+
+bcr_start=23179704
+bcr_end=23318037
+abl_start=130713016
+abl_end=130887675
+
+echo "getting read names"
+
+touch r2.fastq
+
+samtools view -H $STARSOLO_BAM > header.txt
+
+samtools view $STARSOLO_BAM "chr22:22179704-24318037" | cat header.txt - | samtools sort -n | samtools view | grep -v "CB:Z:-" | grep -v "UB:Z:-" | awk -v seen='empty' -v current="another" 'BEGIN{OFS="\t"} {
+ current = $1; # $1 is the read id
+ if (current != seen) {
+  {print "@" current ";"$27";"$28"\n"$10"\n+\n"$11};
+ } seen=current
+}' >> r2.fastq
+
+samtools view $STARSOLO_BAM "chr9:129713016-131887675" | cat header.txt - | samtools sort -n | samtools view | grep -v "CB:Z:-" | grep -v "UB:Z:-" | awk -v seen='empty' -v current="another" 'BEGIN{OFS="\t"} {
+ current = $1; # $1 is the read id
+ if (current != seen) {
+  {print "@" current ";"$27";"$28"\n"$10"\n+\n"$11};
+ } seen=current
+}' >> r2.fastq
+
+# for unmapped reads: they don't have the gx tag, so the CB and UB will be different
+
+samtools view -f 4 $STARSOLO_BAM | samtools sort -n | samtools view | grep -v "CB:Z:-" | grep -v "UB:Z:-" | awk -v seen='empty' -v current="another" 'BEGIN{OFS="\t"} {
+ current = $1; # $1 is the read id
+ if (current != seen) {
+  {print "@" current ";"$22";"$23"\n"$10"\n+\n"$11};
+ } seen=current
+}' >> r2.fastq
+
+echo "finished getting read names"
+
+rm header.txt
+
+# deduplicate .fastqs
+
+cat r2.fastq | paste - - - - | sort -k1,1 -t " " | uniq | tr "\t" "\n" > sorted_duplicate_r2.fastq
```

Comment:
not needed? (as other "not needed" please remove if not)

Reply:
Removed from file